src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -198,25 +253,71 @@ public AutoCompletionCandidates doAutoCompleteJobName(@QueryParameter String val
          * Fills in the available promotion processes.
          */
         public ListBoxModel doFillProcessItems(@AncestorInPath Job defaultJob, @QueryParameter("jobName") String jobName) {
+            final ListBoxModel r = new ListBoxModel();
+            final Jenkins jenkins = Jenkins.getInstance();
+            if (jenkins == null) {
+                return r;
+            }
             if (!defaultJob.hasPermission(Item.CONFIGURE) && defaultJob.hasPermission(Item.EXTENDED_READ)) {
-                return new ListBoxModel();
+                return r;
             }
             defaultJob.checkPermission(Item.CONFIGURE);
 
             AbstractProject<?,?> j = null;
-            if (jobName!=null)
-                j = Jenkins.getInstance().getItem(jobName,defaultJob,AbstractProject.class);
-
-            ListBoxModel r = new ListBoxModel();
+            if (jobName != null) {
+                // JENKINS-25011: also look for jobs in folders.
+                j = getItemByPath(jobName, defaultJob, AbstractProject.class);
+            }
+           
             if (j!=null) {
                 JobPropertyImpl pp = j.getProperty(JobPropertyImpl.class);
                 if (pp!=null) {
-                    for (PromotionProcess proc : pp.getActiveItems())
+                    for (PromotionProcess proc : pp.getActiveItems()) {
+                        // Note: why not list all items instead of active ones?
+                        // this would allow to configure the job even
+                        // if a promotion hasn't happened (yet).
                         r.add(new Option(proc.getDisplayName(),proc.getName()));
+                    }
                 }
             }
             return r;
         }
     }
-
+    
+    //TODO: get relative paths based on folders?
+    @CheckForNull
+    @SuppressWarnings("unchecked")
+    private static <T extends Item> T getItemByPath(@Nonnull String path, @CheckForNull Item currentItem, Class<T> type) {
+        final Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins == null) {
+            return null;
+        }
+        
+        // Legacy behavior
+        TopLevelItem topLevelItem = jenkins.getItem(path);
+        if (topLevelItem != null && type.isAssignableFrom(topLevelItem.getClass())) {
+            return (T)topLevelItem;
+        }
+        
+        // Explicit global addressing
+        if (path.startsWith("/")) {
+            return jenkins.getItemByFullName(path.substring(1), type);
+        }
+        
+        // Explicit relative addressing
+        if (path.startsWith("./")) {
+            String effectivePath = path.substring(2);
+            return jenkins.getItemByFullName(currentItem.getFullName() + "/" + effectivePath, type);
+        }
+        
+        // Emplicit relative addressing
+        T relativeItem = jenkins.getItemByFullName(currentItem.getFullName() + "/" + path, type);
```

Comment:
Given how many programming languages implement `Foo` (in include and similar commands) to mean the sibling of the current element, I think this will be very surprising to a lot of users.

We may well need to distinguish whether the referencing item is an `ItemGroup` or not here, e.g. if it's an `ItemGroup`, `Foo` refers to a child, not a sibling. If it's not, `Foo` refers to a sibling.